### PR TITLE
fix(hunyuan-video): align SGLang and trainside rollouts

### PR DESCRIPTION
**Pull request by leviking98z-rgb**
## Summary

HunyuanVideo's trainside and SGLang recipes used the same visible sampling values but did not execute the same numerical path:

- SGLang filtered rollout intent to `ServerArgs` fields before construction, so Hunyuan's pipeline-only `embedded_cfg_scale` and `flow_shift` were dropped and silently fell back to 6/7 instead of the recipe's 1/5. Route intent through SGLang's `ServerArgs.from_dict` so model `PipelineConfig` fields are preserved.
- Trainside sampling consumed rank-local SDE noise while SGLang derived it per sample and denoising step. Pass the same sample IDs and base seed through Hunyuan's trainside diffusion stage.
- Both bf16-loaded recipes now keep trainable LoRA masters in fp32, while forward compute remains bf16.

The existing `TensorWeightSync + lora_merged` path and `old_logp_source: rollout` recipe are intentionally preserved.

## Related Issue

N/A

## Test Plan

- `SKIP=no-commit-to-branch pre-commit run --files examples/diffusion/hunyuan_video/hunyuan_video_t2v_sglang.yaml examples/diffusion/hunyuan_video/hunyuan_video_t2v_trainside.yaml unirl/models/hunyuan_video/diffusion.py unirl/models/hunyuan_video/pipeline.py unirl/rollout/engine/sglang_diffusion/backends/native.py` — passed.
- `python lint/check_docstring_lines.py` and `python lint/check_recipe_targets.py` — passed.
- Hydra config composition on commit `0b7b0f0`:
  - `python -m unirl.train_diffusion --config-name=diffusion/hunyuan_video/hunyuan_video_t2v_sglang --cfg job --resolve` — passed; resolved `master_dtype=fp32`, `embedded_cfg_scale=1.0`, `flow_shift=5.0`, and `old_logp_source=rollout`.
  - `python -m unirl.train_diffusion --config-name=diffusion/hunyuan_video/hunyuan_video_t2v_trainside --cfg job --resolve` — passed; resolved `master_dtype=fp32`.
- Temporary CPU contract harness on commit `0b7b0f0` — passed. It covers generator forwarding through `step`/`step_with_logp`, per-sample/per-step trainside SDE seeds, seed-count validation, SGLang `PipelineConfig` routing, and the older-runtime fallback. The one-off harness is not committed per repository policy.
- Paired 125-rollout validation: `hunyuanvideo-community/HunyuanVideo`, PickScore prompts, one node / 8 NVIDIA H20 GPUs per run, LoRA rank 64, 1280x720x5, 10 denoising steps, four SDE steps, and `torch_sdpa` for SGLang:
  - [trainside fp32-master reference](https://wandb.ai/leviking98z-zhejiang-university/tmp/runs/hvtsfp32m-8c42b8c)
  - [SGLang lora-merged + rollout-anchor candidate](https://wandb.ai/leviking98z-zhejiang-university/tmp/runs/hvsgmroll32-14e0003)
  - Across steps 1–125, SGLang minus trainside mean reward was `-0.002381`, RMSE was `0.008345`, and correlation was `0.965214`. The original historical pair had RMSE `0.038154` and correlation `0.353175` over the same window.
  - The long run used the functionally equivalent ablation commit `14e0003` with its optional fixed padding disabled. The exact PR commit was then rebased onto current `main` and covered by the composition and CPU contract checks above.

## Compatibility / Risk

The recipe change adds fp32 storage only for trainable LoRA masters; forward mixed precision remains bf16. Deterministic trainside SDE generators are enabled only when the driver supplied initial noise, matching the SGLang path; `disable_driver_xt` retains the previous engine-local RNG behavior. Current SGLang builds use `ServerArgs.from_dict`; older builds without it retain the previous filtered `from_kwargs` path.

## Reviewer Notes

This PR deliberately does not add online-LoRA sync, replay anchoring, fixed text padding, or gradient clipping. The 125-step matrix showed the existing lora-merged + rollout-anchor recipe was the best late-window match.

Duplicate-work check: no open PR or issue covers HunyuanVideo trainside/SGLang reward-curve parity; issue #127 is the general Video RL roadmap. AI assistance was used; the full diff, config composition, contract harness, and long-window metrics were reviewed before publication.

## Checklist

- [x] I reviewed the changed code and removed unrelated/generated artifacts.
- [x] I updated tests, docs, and configs where needed, or explained why not.

### CODE REVIEW

**Review by CjhHa1 — APPROVED** [review]
LGTM

**Comment by CjhHa1** [comment]
Post-merge validation on PR head `7e4106b` (2026-08-17): a paired 20-rollout smoke completed successfully on 2×8 H20, with one 8-GPU trainside arm and one 8-GPU SGLang arm.

Configuration: HunyuanVideo, 1280×720×5, 10 denoising steps, 4 SDE steps, guidance/embedded CFG 1.0, flow shift 5.0, seed 42, and the same prompt stream.

Results:
- trainside: 20/20, rc=0, mean reward `0.735845`
- SGLang: 20/20, rc=0, mean reward `0.731120`
- paired SGLang − trainside mean: `-0.004725`
- paired RMSE: `0.005832`
- Pearson correlation: `0.979853`
- max absolute paired difference: `0.009500`

No rollout-parity issue was observed in this short smoke; the result is consistent with the longer validation reported in the PR description.

Environment note: the pinned SGLang activation JIT header failed to compile in the NeMo image under both system CUDA 12.9 and a CUDA 13.0 compiler. For this smoke only, `silu_and_mul` was routed to the installed precompiled `sgl_kernel` op (validated against the PyTorch fp16 reference on H20, max error `0.00098`). This was a runtime-only environment workaround, not a PR source change. W&B was disabled and PickScore was staged to node-local storage after initialization I/O proved slow.
